dex/src/main/scala/com/wavesplatform/dex/actors/address/BatchOrderCancelActor.scala

```diff
@@ -9,40 +9,48 @@ import com.wavesplatform.dex.domain.order.Order
 import com.wavesplatform.dex.domain.utils.ScorexLogging
 import com.wavesplatform.dex.error
 
+import scala.collection.immutable.ListMap
 import scala.concurrent.duration.FiniteDuration
 
 class BatchOrderCancelActor private (
-  orderIds: Set[Order.Id],
+  orderIds: List[Order.Id],
   source: Command.Source,
   processorActor: ActorRef,
   clientActor: ActorRef,
-  timeout: FiniteDuration,
-  initResponse: Map[Order.Id, OrderCancelResult]
+  timeout: FiniteDuration
 ) extends Actor
     with ScorexLogging {
 
   import BatchOrderCancelActor._
   import context.dispatcher
 
-  orderIds.foreach(processorActor ! CancelOrder(_, source))
+  final private val uniqueOrderIds: Set[Order.Id] = orderIds.toSet
+  uniqueOrderIds.foreach(id => processorActor ! CancelOrder(id, source))
 
-  override def receive: Receive = state(orderIds, initResponse, context.system.scheduler.scheduleOnce(timeout, self, TimedOut))
+  override def receive: Receive = state(uniqueOrderIds, Map.empty, context.system.scheduler.scheduleOnce(timeout, self, TimedOut))
 
   private def state(restOrderIds: Set[Order.Id], response: Map[Order.Id, OrderCancelResult], timer: Cancellable): Receive = {
+
     case CancelResponse(id, x) =>
       val updatedRestOrderIds = restOrderIds - id
       val updatedResponse = response.updated(id, x)
 
-      if (updatedRestOrderIds.isEmpty) stop(Event.BatchCancelCompleted(updatedResponse), timer)
+      if (updatedRestOrderIds.isEmpty)
+        stop(Event.BatchCancelCompleted(mkSortedResponse(updatedResponse)), timer)
       else context.become(state(restOrderIds - id, updatedResponse, timer))
 
     // case Terminated(ref) => // Can't terminate before processorActor, because processorActor is a parent
 
     case TimedOut =>
       log.error(s"CancelOrder is timed out for orders: ${restOrderIds.mkString(", ")}")
-      stop(Event.BatchCancelCompleted(response), timer)
+      stop(Event.BatchCancelCompleted(mkSortedResponse(response)), timer)
   }
 
+  private def mkSortedResponse(response: Map[Order.Id, OrderCancelResult]): ListMap[Order.Id, OrderCancelResult] = (for {
```

Comment:
`mkOrderedResponse`, because we didn't sort elements, we order them by the order in `orderIds`

Reply:
ok. fixed